docs/kvbc/block_merkle_category.md

```diff
@@ -0,0 +1,359 @@
+tree version `X+1` and a new leaf node being created.
+
+![Sparse Merkle Update Example](img/SparseMerkleUpdate.svg)
+
+Such a persistent (copy-on-write) tree has the benefit that a user can have a signed copy of the
+merkle root at version `X` and still prove that Leaf `OF1B` was part of that version of the tree.
+With a mutable tree, a user could only prove that that a key is part of the latest version of the
+tree. If a key is overwritten, and there is no old copy of the tree, the user cannot prove that
+the prior version ever existed if only ****given**** the current root hash.
+
+However, the mutable tree has the advantage that it requires much less storage overhead than a
+persistent tree. In our persistent version, we have to rewrite a new root node, node at `0` and
+node at `0F`, rather than simply updating the existing nodes. This is 3 extra
+BatchedInternalNodes. In a large tree, the top level nodes will be nearly full (31 hashes +
+metadata), with a full node taking up approximately 1kb of space. Space analysis will be given in
+a section below, but for billions of keys the overhead is **a lot**. Importantly, the overhead
+gets worse as more keys are inserted, rather than better. This storage overhead is expensive, and
+it also significantly impacts performance as much more data is read and written than if the keys
+and values were simply stored alone.
+
+What is needed is a way to get the bulk of the benefits of the sparse merkle tree, while limiting
+the overhead. The solution should also scale well: more keys should not cause overhead to
+increase significantly. Our solution is to reuse the sparse merkle tree, but to only store
+per-block data, rather than per-key data. The more keys we batch into a block, the lower the
+overhead of the merkle tree.
+
+The block based usage of the sparse merkle tree is part of a process of **storage
+categorization** in KVBC, where different keys can be stored in different **categories**, with
+each category providing different guarantees. The work described here is part of the
+`BlockMerkle` category.
+
+Categorization in general, and other key categories, in particular, are out of scope for this
+document. It is still worth being aware that this is just one of many ways to store data in KVBC.
+Users of concord-bft do not have to use the `BlockMerkle` category, nor do they have to use KVBC
+at all. The goals and assumptions below are justifications for when a user may want to consider
+using the `BlockMerkle` category.
+
+
+# Goals
+
+* Reduce the overhead of the merkle tree in terms of performance and storage capacity
+* Maintain the ability to prune blocks and keys and still reference and prove **active** keys created in blocks that have been pruned
+* Maintain the ability to lookup keys directly without traversing the merkle tree
+* Provide the ability to scale the merkle tree with block size such that it works with batching.
```

Comment:
Nit: dot at the end and no dots on lines above.

Reply:
I deleted the section in the implementation doc.